### PR TITLE
F/page break extend

### PR DESCRIPTION
**Pull request by dev7ch**
https://wepublish.atlassian.net/browse/BAJ-191

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ yarn-error.log
 .env
 .changelog
 coverage/
+lerna-debug.log
```
Comment (roosnic1) [inline .gitignore:16]: remove

File: examples/website/src/shared/blocks/blockRenderer.tsx [hunk 2]
```diff
@@ -97,9 +97,16 @@ export function renderBlock(block: Block | null, opts: RenderBlockOptions) {
       return (
         <PageBreakBlock
           text={block.value.text}
+          richText={[]}
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/blockRenderer.tsx:100]: why not  `block.value.richText`?

File: examples/website/src/shared/blocks/blockRenderer.tsx [hunk 3]
```diff
@@ -97,9 +97,16 @@ export function renderBlock(block: Block | null, opts: RenderBlockOptions) {
...
           linkURL={block.value.linkURL}
+          styleOption={block.value.styleOption}
+          layoutOption={block.value.layoutOption}
+          templateOption={block.value.layoutOption}
           linkText={block.value.linkText}
+          linkTarget={block.value.linkTarget}
+          imageID={block.value.imageID}
           isArticle={isArticle}
+          image={null}
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/blockRenderer.tsx:109]: Why `image` and `imageID`?

File: examples/website/src/shared/blocks/peerPageBreak.stories.tsx [hunk 4]
```diff
@@ -10,15 +10,29 @@ storiesOf('Blocks|PeerPageBreak*', module)
   .add('Default', () => (
     <PageBreakBlock
       text={'Unterstuetze uns!'}
+      richText={[]}
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/peerPageBreak.stories.tsx:13]: We don't update the stories in the examples website.

File: examples/website/src/shared/blocks/peerPageBreak.tsx [hunk 5]
```diff
@@ -6,6 +6,7 @@ import {cssRule, useStyle} from '@karma.run/react'
 import {Color} from '../style/colors'
 import {pxToRem, whenMobile} from '../style/helpers'
 import {Link} from '../route/routeContext'
+import {RichTextBlockValue} from '@wepublish/editor/src/client/blocks/types'
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/peerPageBreak.tsx:9]: No cross package imports

File: examples/website/src/shared/blocks/peerPageBreak.tsx [hunk 6]
```diff
@@ -78,6 +93,15 @@ export function PageBreakBlock({
             <Image src={peer.logoURL} height={90} width={90} />
           </div>
         )}
+        <pre>
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/peerPageBreak.tsx:96]: That's not really useful rather just leave it as it was

File: examples/website/src/shared/blocks/peerPageBreak.tsx (hunk 5 above)
Comment (roosnic1) [reply examples/website/src/shared/blocks/peerPageBreak.tsx:9]: Checkout how the richText Block handles this

File: examples/website/src/shared/blocks/peerPageBreak.tsx [hunk 7]
```diff
@@ -52,23 +53,37 @@ const PeerPageBreakButton = cssRule({
...
   text: string
+  richText: RichTextBlockValue
   linkURL: string
   linkText: string
+  linkTarget?: string
+  styleOption?: string
+  templateOption?: string
+  layoutOption?: string
+  imageID?: string | undefined
+  image?: null
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/peerPageBreak.tsx:70]: again why both `image` and `imageID`?

File: examples/website/src/shared/blocks/peerPageBreak.tsx [hunk 8]
```diff
@@ -52,23 +53,37 @@ const PeerPageBreakButton = cssRule({
...
   text,
+  richText,
   linkURL,
   linkText,
+  linkTarget,
+  styleOption,
+  layoutOption,
+  templateOption,
+  imageID,
+  image,
```
Comment (roosnic1) [inline examples/website/src/shared/blocks/peerPageBreak.tsx:84]: again why both `image` and `imageID`?

File: examples/website/src/shared/route/blockAdapters.ts [hunk 9]
```diff
@@ -90,8 +90,14 @@ function getBlocks(blocks: any, articleMeta?: ArticleMeta): Block[] {
...
             text: block.text,
+            richText: block.richText,
             linkURL: block.linkURL,
-            linkText: block.linkText
+            linkText: block.linkText,
+            linkTarget: block.linkTarget,
+            styleOption: block.styleOption,
+            layoutOption: block.layoutOption,
+            imageID: block.image?.id,
+            image: block.image && imageAdapter(block.image)
```
Comment (roosnic1) [inline examples/website/src/shared/route/blockAdapters.ts:100]: again why both `image` and `imageID`?

File: examples/website/src/shared/route/blockAdapters.ts [hunk 10]
```diff
@@ -111,7 +117,7 @@ function getBlocks(blocks: any, articleMeta?: ArticleMeta): Block[] {
           title: block.title,
           lead: block.lead
         }
-        if (articleMeta && (index == 0 || (hasTitleImage && index == 1))) {
+        if (articleMeta && (index === 0 || (hasTitleImage && index === 1))) {
```
Comment (roosnic1) [inline examples/website/src/shared/route/blockAdapters.ts:120]: This change doesn't really belong to this PR or why did you change this?

File: examples/website/src/shared/route/gqlFragments.ts [hunk 11]
```diff
@@ -294,8 +294,16 @@ export const linkPageBreakBlockDataFragment = gql`
   fragment LinkPageBreakBlockData on LinkPageBreakBlock {
     __typename
     text
+    htmlText
```
Comment (roosnic1) [inline examples/website/src/shared/route/gqlFragments.ts:297]: Isn't it `richText`?

File: examples/website/src/shared/route/gqlFragments.ts [hunk 12]
```diff
@@ -294,8 +294,16 @@ export const linkPageBreakBlockDataFragment = gql`
...
     linkURL
     linkText
+    hideButton
```
Comment (roosnic1) [inline examples/website/src/shared/route/gqlFragments.ts:300]: aren't you missing the `templateOption` here?

File: examples/website/src/shared/types.ts [hunk 13]
```diff
@@ -1,5 +1,6 @@
 import {Node} from 'slate'
 import {Route} from './route/routeContext'
+import {RichTextBlockValue} from '@wepublish/editor/src/client/blocks/types'
```
Comment (roosnic1) [inline examples/website/src/shared/types.ts:3]: same as above no cross imports

File: examples/website/src/shared/types.ts [hunk 14]
```diff
@@ -156,8 +157,14 @@ export interface PeerPageBreakBlock
...
       peer: Peer
       text: string
+      richText: RichTextBlockValue
       linkURL: string
       linkText: string
+      linkTarget: string
+      styleOption: string
+      layoutOption: string
+      imageID: string
+      image: undefined
```
Comment (roosnic1) [inline examples/website/src/shared/types.ts:167]: again why both `image` and `imageID`?

File: lerna.json [hunk 15]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.8.5",
+  "version": "0.8.24",
```
Comment (roosnic1) [inline lerna.json:2]: nope

File: lerna.json [hunk 16]
```diff
@@ -16,7 +16,10 @@
   "npmClient": "yarn",
   "command": {
     "version": {
-      "allowBranch": "master"
+      "allowBranch": [
```
Comment (roosnic1) [inline lerna.json:19]: remove. I'll have a task for this soon

File: lerna.json [hunk 17]
```diff
@@ -29,7 +32,7 @@
   "packages": [
     "packages/api",
     "packages/api-db-mongodb",
-    "packages/api-media-karma",
-    "packages/editor"
+    "packages/editor",
```
Comment (roosnic1) [inline lerna.json:35]: why though? :)

File: packages/api-db-karma/src/index.ts [hunk 18]
```diff
@@ -372,11 +372,7 @@ export class KarmaStorageAdapter implements StorageAdapter {
       define('article', all(tag(ModelTag.Article))
         .filterList((index, value) => value.field('id').equal(string(id)))
         .first()),
-      define('articleVersion', scope('article')
-        .field('versions')
-        .slice(version, 1)
-        .first()
-        .get()),
+      define('articleVersion', scope('article').field('versions').slice(version, 1).first().get()),
```
Comment (roosnic1) [inline packages/api-db-karma/src/index.ts:375]: please revert all changes in this file

File: packages/api-db-mongodb/CHANGELOG.md [hunk 19]
```diff
@@ -3,6 +3,18 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+
```
Comment (roosnic1) [inline packages/api-db-mongodb/CHANGELOG.md:6]: please revert the changes in this file also

File: packages/api-db-mongodb/src/migration.ts [hunk 20]
```diff
@@ -277,6 +277,68 @@ export const Migrations: Migration[] = [
...
+  },
+  {
+    // Add RTE to page break block if not exists in articles and pages
+    version: 4,
+    async migrate(db) {
+      await db.collection(CollectionName.Articles).updateMany(
+        {'draft.blocks': {$elemMatch: {type: 'linkPageBreak', richText: {$exists: false}}}},
+        {
+          $set: {
+            'published.blocks.$[elem].richText': [{children: [{text: ''}], type: 'paragraph'}]
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/migration.ts:289]: i guess this should be `draft.blocks`

File: packages/api-db-mongodb/src/migration.ts [hunk 21]
```diff
@@ -277,6 +277,68 @@ export const Migrations: Migration[] = [
...
+        {$set: {'pending.blocks.$[elem].richText': [{children: [{text: ''}], type: 'paragraph'}]}},
+        {arrayFilters: [{'elem.type': 'linkPageBreak'}]}
+      )
+
+      // Add RTE to page break block if not exists in pages
+      await db.collection(CollectionName.Pages).updateMany(
+        {'draft.blocks': {$elemMatch: {type: 'linkPageBreak', richText: {$exists: false}}}},
+        {
+          $set: {
+            'published.blocks.$[elem].richText': [{children: [{text: ''}], type: 'paragraph'}]
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/migration.ts:318]: i guess this should also be `draft.blocks`

File: packages/api/__tests__/api/private/block.graphql [hunk 22]
```diff
@@ -74,6 +74,15 @@ fragment FullBlock on Block {
     text
     linkText
     linkURL
+    linkTarget
```
Comment (roosnic1) [inline packages/api/__tests__/api/private/block.graphql:77]: You don't need to update the tests if you don't intend to test the changes.

File: packages/api/__tests__/api/public/block.graphql [hunk 23]
```diff
@@ -72,8 +72,17 @@ fragment FullBlock on Block {
 
   ... on LinkPageBreakBlock {
     text
+    richText
```
Comment (roosnic1) [inline packages/api/__tests__/api/public/block.graphql:75]: You don't need to update the tests if you don't intend to test the changes.

File: packages/api/src/db/block.ts [hunk 24]
```diff
@@ -102,8 +102,15 @@ export interface ListicleBlock {
 export interface LinkPageBreakBlock {
   type: BlockType.LinkPageBreak
   text: string
+  richText: RichTextNode[]
```
Comment (roosnic1) [inline packages/api/src/db/block.ts:105]: Since you need a migration for the `LinePageBreakBlock` i suggest that you replace the `text` field with the `richText`. I don't see any reason why anyone needs both.

File: packages/api/src/db/block.ts [hunk 25]
```diff
@@ -102,8 +102,15 @@ export interface ListicleBlock {
...
   linkURL: string
   linkText: string
+  linkTarget: string
+  hideButton: boolean
```
Comment (roosnic1) [inline packages/api/src/db/block.ts:109]: I don't see the point of this boolean. Can't you just make `linkText` optional and if it's empty don't display a button?

File: packages/api/src/graphql/blocks.ts [hunk 26]
```diff
@@ -442,8 +443,20 @@ export const GraphQLLinkPageBreakBlock = new GraphQLObjectType<LinkPageBreakBloc
   name: 'LinkPageBreakBlock',
   fields: {
     text: {type: GraphQLString},
+    richText: {type: GraphQLNonNull(GraphQLRichText)},
     linkURL: {type: GraphQLString},
-    linkText: {type: GraphQLString}
+    linkText: {type: GraphQLString},
+    linkTarget: {type: GraphQLString},
+    hideButton: {type: GraphQLBoolean},
```
Comment (roosnic1) [inline packages/api/src/graphql/blocks.ts:450]: i would make this also `GraphQLNonNull`

File: packages/api/src/graphql/blocks.ts [hunk 27]
```diff
@@ -540,8 +553,15 @@ export const GraphQLLinkPageBreakBlockInput = new GraphQLInputObjectType({
   name: 'LinkPageBreakBlockInput',
   fields: {
     text: {type: GraphQLString},
+    richText: {type: GraphQLNonNull(GraphQLRichText)},
     linkURL: {type: GraphQLString},
-    linkText: {type: GraphQLString}
+    linkText: {type: GraphQLString},
+    linkTarget: {type: GraphQLString},
+    hideButton: {type: GraphQLBoolean},
```
Comment (roosnic1) [inline packages/api/src/graphql/blocks.ts:560]: i would make this also `GraphQLNonNull`

File: packages/editor/package.json [hunk 28]
```diff
@@ -59,7 +59,7 @@
     "core-js": "^3.4.8",
     "express": "^4.17.1",
     "graphql": "^14.5.8",
-    "graphql-tag": "^2.10.1",
+    "graphql-tag": "^2.11.0",
```
Comment (roosnic1) [inline packages/editor/package.json:62]: why the update?

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx [hunk 29]
```diff
@@ -13,46 +34,218 @@ export function LinkPageBreakBlock({
...
+    templateOption
+  } = value
   const focusRef = useRef<HTMLTextAreaElement>(null)
+  const focusInputRef = useRef<HTMLInputElement>(null)
+
+  /* eslint-disable i18next/no-literal-string */
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
 
-  const {t} = useTranslation()
+  // const {t} = useTranslation()
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/linkPageBreakBlock.tsx:55]: Please add all language translations to the `en.json`

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx [hunk 30]
```diff
@@ -13,46 +34,218 @@ export function LinkPageBreakBlock({
...
+    richText,
+    linkTarget,
+    hideButton,
+    image,
+    templateOption
+  } = value
   const focusRef = useRef<HTMLTextAreaElement>(null)
+  const focusInputRef = useRef<HTMLInputElement>(null)
+
+  /* eslint-disable i18next/no-literal-string */
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/linkPageBreakBlock.tsx:52]: Nope.

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx [hunk 31]
```diff
@@ -13,46 +34,218 @@ export function LinkPageBreakBlock({
...
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/linkPageBreakBlock.tsx:53]: If you really need this please add a comment why.

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx [hunk 32]
```diff
@@ -13,46 +34,218 @@ export function LinkPageBreakBlock({
...
-      <TypographicTextArea
-        ref={focusRef}
-        variant="subtitle2"
-        placeholder={t('blocks.linkPageBreak.linkURL')}
-        align="center"
-        value={linkURL}
-        disabled={disabled}
-        onChange={e => onChange({...value, linkURL: e.target.value})}
-      />
+      <div style={{display: 'flex', flexFlow: 'row wrap'}}>
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/linkPageBreakBlock.tsx:75]: My main problem with this block is that we loose the `WYSIWYG` approach for this block. Since you need to touch this part again because of the UI change i suggest that you move all the Selects, booleans and Image selection to a Drawer. You can use the `embedEditPanel.tsx` as an inspiration.

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: .env.example [hunk 33]
```diff
@@ -0,0 +1 @@
+GH_TOKEN=<personal-github-token-to-release-with-lerna>
```
[the remark this answers is not in the document]
Comment (dev7ch) [reply .env.example:1]: its missing in the codebase, why?

File: .gitignore (hunk 1 above)
Comment (dev7ch) [reply .gitignore:16]: why this should be in vcs? does it harm?

File: examples/website/src/shared/blocks/blockRenderer.tsx (hunk 2 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/blockRenderer.tsx]: corrected

File: examples/website/src/shared/blocks/peerPageBreak.tsx (hunk 5 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.tsx]: corrected

File: examples/website/src/shared/blocks/peerPageBreak.tsx (hunk 7 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.tsx]: i updated this to show selected values

File: examples/website/src/shared/route/blockAdapters.ts (hunk 9 above)
Comment (dev7ch) [reply examples/website/src/shared/route/blockAdapters.ts:101]: its from before, i did not updated all examples, corrected

File: examples/website/src/shared/blocks/peerPageBreak.tsx (hunk 8 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.tsx:85]: its from before, i did not updated all examples, corrected

File: examples/website/src/shared/route/gqlFragments.ts (hunk 11 above)
Comment (dev7ch) [reply examples/website/src/shared/route/gqlFragments.ts]: yes, thanks

File: examples/website/src/shared/route/gqlFragments.ts (hunk 12 above)
Comment (dev7ch) [reply examples/website/src/shared/route/gqlFragments.ts:300]: yes, corrected

File: examples/website/src/shared/types.ts (hunk 13 above)
Comment (dev7ch) [reply examples/website/src/shared/types.ts]: corrected

File: examples/website/src/shared/types.ts (hunk 14 above)
Comment (dev7ch) [reply examples/website/src/shared/types.ts]: as above

File: packages/api-db-karma/src/index.ts (hunk 18 above)
Comment (dev7ch) [reply packages/api-db-karma/src/index.ts:375]: this was applied by `yarn prettier` why remove? or better exclude unmaintained from liniting?

File: packages/api-db-mongodb/src/migration.ts (hunk 20 above)
Comment (dev7ch) [reply packages/api-db-mongodb/src/migration.ts]: yes, thanks, corrected

File: packages/api/__tests__/api/private/block.graphql (hunk 22 above)
Comment (dev7ch) [reply packages/api/__tests__/api/private/block.graphql:77]: yep

Comment (dev7ch) [reply packages/api/__tests__/api/private/block.graphql:77]: would you like to add tests? does it harm?

File: packages/api/src/db/block.ts (hunk 24 above)
Comment (dev7ch) [reply packages/api/src/db/block.ts:105]: the `text` is basically the title, the migration was only needed for RTE field and would bring conflicts, so better no field removal / migrations

File: packages/api/src/db/block.ts (hunk 25 above)
Comment (dev7ch) [reply packages/api/src/db/block.ts:109]: its an options, which makes the whole element clickable, it keeps the value of the linkText label to reactivate, to make it more understandable and flexible it better to have a toggler with diescription

File: packages/api/src/graphql/blocks.ts (hunk 26 above)
Comment (dev7ch) [reply packages/api/src/graphql/blocks.ts:450]: why? its a boolean

File: packages/api/src/graphql/blocks.ts (hunk 27 above)
Comment (dev7ch) [reply packages/api/src/graphql/blocks.ts:560]: saa

File: packages/editor/package.json (hunk 28 above)
Comment (dev7ch) [reply packages/editor/package.json:62]: in the yarn.log a found 2 versions, i think it was the api which used `2.11.0` just tu use the same version on all pakcages

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 29 above)
Comment (dev7ch) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:55]: could you or your team do it?

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 30 above)
Comment (dev7ch) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:52]: UI replacement next days/weeks??

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 32 above)
Comment (dev7ch) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:75]: ok, but the current panel UX if not really satisfying (more clicks, no close on overlay click, etc) and the will be a UI replace. The current UX is more easy and faster to use for editors so i suggest to just keep it as it is as you ll touch it in the UI replacement process

once the panel actions are more user friendly this can be reranged better.

**Comment by dev7ch** [comment]
image
image

File: packages/api/__tests__/api/public/block.graphql (hunk 23 above)
Comment (dev7ch) [reply packages/api/__tests__/api/public/block.graphql:75]: does it harm?

File: examples/website/src/shared/blocks/peerPageBreak.stories.tsx (hunk 4 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.stories.tsx:13]: so can i removed it completely?

File: .env.example (hunk 33 above)
[the remark this answers is not in the document]
Comment (roosnic1) [reply .env.example:1]: The `.env.example` files are used in the setup command of the individual packages. There is no plan yet to create a root `.env.example`.

File: .gitignore (hunk 1 above)
Comment (roosnic1) [reply .gitignore:16]: it doesn't belong to the scope of the PR.

File: examples/website/src/shared/blocks/peerPageBreak.stories.tsx (hunk 4 above)
Comment (roosnic1) [reply examples/website/src/shared/blocks/peerPageBreak.stories.tsx:13]: yes

File: examples/website/src/shared/route/blockAdapters.ts (hunk 9 above)
Comment (roosnic1) [reply examples/website/src/shared/route/blockAdapters.ts:101]: Will you change this?

File: packages/api-db-karma/src/index.ts (hunk 18 above)
Comment (roosnic1) [reply packages/api-db-karma/src/index.ts:375]: just remove it from the PR.

File: packages/api/__tests__/api/private/block.graphql (hunk 22 above)
Comment (roosnic1) [reply packages/api/__tests__/api/private/block.graphql:77]: Not yet ready for testing blocks.

File: packages/api/src/db/block.ts (hunk 24 above)
Comment (roosnic1) [reply packages/api/src/db/block.ts:105]: In the `richText` you can have title and text in one. The `richText` field makes the `text` field redundant. And since you're migrating the blocks already it's an easy task to migrate the `text` field to the `richText` field.

File: packages/api/src/graphql/blocks.ts (hunk 26 above)
Comment (roosnic1) [reply packages/api/src/graphql/blocks.ts:450]: Yes. That's why. A boolean should never be null.

File: packages/api/src/db/block.ts (hunk 25 above)
Comment (roosnic1) [reply packages/api/src/db/block.ts:109]: How does that work with links inside the richText editor?

File: packages/api/src/graphql/blocks.ts (hunk 27 above)
Comment (roosnic1) [reply packages/api/src/graphql/blocks.ts:560]: Yes. That's why. A boolean should never be null.

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 29 above)
Comment (roosnic1) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:55]: Nope.

File: packages/editor/package.json (hunk 28 above)
Comment (roosnic1) [reply packages/editor/package.json:62]: Not part of this PR.

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 32 above)
Comment (roosnic1) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:75]: The UI platform works with the same panel concept. For the forseable future we'll stick with that. New UI branch is almost ready. Will be merged this week. That means you can already merge it into your branch and replace the input elements and move them to the panel.

File: packages/api/__tests__/api/public/block.graphql (hunk 23 above)
Comment (roosnic1) [reply packages/api/__tests__/api/public/block.graphql:75]: if it doesn't do anything don't add it.

File: examples/website/src/shared/route/blockAdapters.ts (hunk 10 above)
Comment (dev7ch) [reply examples/website/src/shared/route/blockAdapters.ts:121]: its basically a mistake, but will revert for now

File: examples/website/src/shared/blocks/peerPageBreak.tsx (hunk 6 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.tsx]: updated

File: examples/website/src/shared/route/blockAdapters.ts (hunk 9 above)
Comment (dev7ch) [reply examples/website/src/shared/route/blockAdapters.ts:101]: yes

File: lerna.json (hunk 16 above)
Comment (dev7ch) [reply lerna.json:19]: can you link the ticket and task here please?

File: lerna.json (hunk 17 above)
Comment (dev7ch) [reply lerna.json:35]: copy pasta

File: packages/api/__tests__/api/public/block.graphql (hunk 23 above)
Comment (dev7ch) [reply packages/api/__tests__/api/public/block.graphql:75]: i d suggest to merge and then directky add test with your team to understand the extended block better

File: packages/api/src/db/block.ts (hunk 24 above)
Comment (dev7ch) [reply packages/api/src/db/block.ts:105]: it breaks the layout and concept of the block. Title and Text (here richText) are two fields

File: packages/api/src/db/block.ts (hunk 25 above)
Comment (dev7ch) [reply packages/api/src/db/block.ts:109]: links inside editor are clear text links, this is a button and if you select an image left or right its clickable.

If you use RTE only i assume you cant click the link now (but think the problem you try to find is solvable by css)

File: packages/api/src/graphql/blocks.ts (hunk 26 above)
Comment (dev7ch) [reply packages/api/src/graphql/blocks.ts:450]: does it harm or break the functionality?

Comment (dev7ch) [reply packages/api/src/graphql/blocks.ts:450]: (afik i see in current setup there is no migration needed, for nonNull i think it is)

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 32 above)
Comment (dev7ch) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:75]: Merge this before you chnage the UI, so that you can dont block the reintegration with you UI changes

File: .env.example (hunk 33 above)
[the remark this answers is not in the document]
Comment (dev7ch) [reply .env.example:1]: you the an `.env` fill in root dir to deploy with lerna project agnostic

File: examples/website/src/shared/blocks/peerPageBreak.stories.tsx (hunk 4 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.stories.tsx]: deleted

File: examples/website/src/shared/route/blockAdapters.ts (hunk 9 above)
Comment (dev7ch) [reply examples/website/src/shared/route/blockAdapters.ts:101]: image

Its changed, it seems to be an GH UI bug, i dunno ... see here https://github.com/bajour/wepublish/blob/f/page-break-extend/examples/website/src/shared/route/blockAdapters.ts#L100

File: examples/website/src/shared/blocks/peerPageBreak.tsx (hunk 8 above)
Comment (dev7ch) [reply examples/website/src/shared/blocks/peerPageBreak.tsx:85]: changed, GH UI not updated, see https://github.com/bajour/wepublish/blob/f/page-break-extend/examples/website/src/shared/blocks/peerPageBreak.tsx

File: packages/api/__tests__/api/private/block.graphql (hunk 22 above)
Comment (dev7ch) [reply packages/api/__tests__/api/private/block.graphql:77]: you introduced testing, which is still in development, i suggest you add this to understand the block extension better

File: .gitignore (hunk 1 above)
Comment (dev7ch) [reply .gitignore:16]: its missing

File: .env.example (hunk 33 above)
[the remark this answers is not in the document]
Comment (roosnic1) [reply .env.example:1]: yes and i have one and we don't need an `example`

File: lerna.json (hunk 16 above)
Comment (roosnic1) [reply lerna.json]: Doesn't exist yet.

File: packages/api/__tests__/api/private/block.graphql (hunk 22 above)
Comment (roosnic1) [reply packages/api/__tests__/api/private/block.graphql:77]: Just remove it.

File: packages/api/__tests__/api/public/block.graphql (hunk 23 above)
Comment (roosnic1) [reply packages/api/__tests__/api/public/block.graphql:75]: just remove it from the PR

File: packages/api/src/graphql/blocks.ts (hunk 26 above)
Comment (roosnic1) [reply packages/api/src/graphql/blocks.ts:450]: What are you talking about? I'd like you to updat ethe GraphQL schema to make the boolean a NonNull since a boolean is always true or false.

File: packages/editor/src/client/blocks/linkPageBreakBlock.tsx (hunk 32 above)
Comment (roosnic1) [reply packages/editor/src/client/blocks/linkPageBreakBlock.tsx:75]: I will merge this once it's ready.

**Comment by roosnic1** [comment]
Closed in favor of #72
